src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -63,120 +64,218 @@ public FetchAndMergeEntry(BibDatabaseContext bibDatabaseContext,
+        );
+    }
+
+    public void fetchAndMergeBatch(List<BibEntry> entries) {
+        entries.forEach(entry -> SUPPORTED_IDENTIFIER_FIELDS.forEach(field -> fetchAndMergeEntry(entry, field)));
+    }
+
+    private void fetchAndMergeEntry(BibEntry entry, Field field) {
+        entry.getField(field)
+             .flatMap(fieldContent -> WebFetchers.getIdBasedFetcherForField(field, preferences.getImportFormatPreferences()))
+             .ifPresent(fetcher -> executeFetchTask(fetcher, field, entry));
+    }
+
+    private void executeFetchTask(IdBasedFetcher fetcher, Field field, BibEntry entry) {
+        entry.getField(field)
+             .ifPresent(fieldContent -> BackgroundTask.wrap(() -> fetcher.performSearchById(fieldContent))
+                                                      .onSuccess(fetchedEntry -> processFetchedEntry(fetchedEntry, field, entry))
+                                                      .onFailure(exception -> handleFetchException(exception, fetcher))
+                                                      .executeWith(taskExecutor));
+    }
+
+    private void processFetchedEntry(Optional<BibEntry> fetchedEntry, Field field, BibEntry originalEntry) {
+        ImportCleanup cleanup = ImportCleanup.targeting(bibDatabaseContext.getMode(), preferences.getFieldPreferences());
+        fetchedEntry.ifPresentOrElse(
+                entry -> {
+                    cleanup.doPostCleanup(entry);
+                    mergeWithoutDialog(originalEntry, entry);
+                },
+                // Notify if no entry was fetched
+                () -> dialogService.notify(Localization.lang("Cannot get info based on given %0: %1", field.getDisplayName(), originalEntry.getField(field).orElse("")))
+        );
+    }
+
+    private void handleFetchException(Exception exception, IdBasedFetcher fetcher) {
+        LOGGER.error("Error while fetching bibliographic information", exception);
+        String fetcherName = fetcher.getName();
+        // Handle different types of exceptions with specific error messages
+        if (exception instanceof FetcherClientException) {
+            dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("No data was found for the identifier"));
+        } else if (exception instanceof FetcherServerException) {
+            dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("Server not available"));
+        } else {
+            dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("Error occurred %0", exception.getMessage()));
+        }
+    }
+
+    private void mergeWithoutDialog(BibEntry originalEntry, BibEntry fetchedEntry) {
```

Comment:
This whole method can be go into `logic`, because there is no gui interaction - and it should be testable with a JUnit test.

Annote the test using JUnit

    @AllowedToUseSwing("Requires undo/redo functionality")

Reply:
I apologize for the oversight; I should have provided my comments after pushing the changes. Over the past few hours, I have been working on optimizing the code further and transitioning the class into the logic directory. 
My current progress is that ''' 

Refactor FetchAndMergeEntry:

The single-entry fetching and merging logic remains in FetchAndMergeEntry.

Extracted logic for batch processing and handling multiple entries from FetchAndMergeEntry.

Introduce MergingIdBasedFetcher:

Created the MergingIdBasedFetcher class in the org.jabref.logic.importer.fetcher package.

MergingIdBasedFetcher handles mass operations such as fetching and merging multiple BibEntry objects.

Uses NotificationService to provide error and status notifications, removing reliance on user confirmation dialogs for better automation.

Create MergeEntriesHelper:

Introduced the MergeEntriesHelper class to centralize and handle the actual merging logic between BibEntry instances.

The class includes methods to update entry types, fields, and remove obsolete fields during merging. '''